src/app/pet-profile/pet-form/pet-form-validators.ts

```diff
@@ -0,0 +1,10 @@
+import { AbstractControl, FormGroup } from "@angular/forms";
+
+// Custom validator function
+export function medicationValidator(control: AbstractControl){
+    const group = control as FormGroup;
+    const medName = group.get('med_name')?.value;
```

Comment:
Please remove all hardcoded strings. Create a "CommonConstants" class and put all common strings there. Then import the "CommonConstants" file and use the constant you need.

This has the following benefits:
- prevents typing errors
- allows autocomplete
- if you ever need to change the string name, you can do so in one place.

E.g. 

```ts
// app-constants.ts
export abstract class CommonConstants {
  static readonly MED_NAME: string = 'med_name';
}
```
```ts
// pet-form-validators.ts
import { CommonConstants } from './app-constants';
const medName = group.get(CommonConstants.MED_NAME)?.value;
````